src/main.rs

```diff
@@ -75,25 +75,31 @@ fn real_main(args: Args) -> anyhow::Result<()> {
     let generator = SbomGenerator { all: args.all };
 
     log::trace!("SBOM generation started");
-    let bom = generator.create_sbom(manifest)?;
+    let boms = generator.create_sboms(manifest)?;
     log::trace!("SBOM generation finished");
 
     match args.format {
         Format::Json => {
             log::info!("Outputting bom.json");
-            serde_json::to_writer_pretty(File::create("bom.json")?, &bom)
-                .map_err(anyhow::Error::from)?;
+            for (manifest_path, bom) in boms {
+                let path = manifest_path.with_file_name("bom.json");
```

Comment:
For this and the XML one below, it would be nice to remove the `log::info!` line two lines above and add a line after this one with

``` rust
log::info!("Outputting {}", path.to_string_lossy());
```

Reply:
"Path::display exists for precisely this purpose so I've used that instead."